test/extended/util/client.go

```diff
@@ -337,6 +345,40 @@ func (c *CLI) SetupProject() string {
 	return newNamespace
 }
 
+func (c *CLI) setupNamespace() string {
+	requiresTestStart()
+	newNamespace := names.SimpleNameGenerator.GenerateName(fmt.Sprintf("e2e-test-%s-", c.kubeFramework.BaseName))
+	c.SetNamespace(newNamespace).ChangeUser(fmt.Sprintf("%s-user", newNamespace))
```

Comment:
1. you should try to execute this and see it work without the user API
2. you could create a new ChangeUserWithClientCert that uses a client CSR to change the identity and persist in a kubeconfig file.

Reply:
It works because of https://github.com/openshift/origin/pull/27358/files#diff-e21320d17fdce22cad99f368c20f8aca8b28bac96f890d69e95fda9be0ae7f93R822
Here, `AdminConfig` resolves to the kubeconfig that was used to launch the tests binary, so its the same as the `UserConfig`.
Anyway I can create a function like you say, will follow that approach.

Reply:
Fixed